apps/vaults/components/details/actions/QuickActionsButtons.tsx

```diff
@@ -0,0 +1,165 @@
+	const isInputTokenEth = selectedOptionFrom?.value === ETH_TOKEN_ADDRESS;
+	const isOutputTokenEth = selectedOptionTo?.value === ETH_TOKEN_ADDRESS;
+	const canInteract = isActive && amount.raw.gt(0) && selectedOptionFrom && selectedOptionTo;
+
+	const spender = useMemo((): TAddress => {
+		if (currentSolver === Solvers.CHAIN_COIN && isOutputTokenEth) {
+			return (toAddress(getEthZapperContract(safeChainID)));
+		} else if (currentSolver === Solvers.PARTNER_CONTRACT) {
+			return (toAddress(networks?.[safeChainID]?.partnerContractAddress));
+		}
+		return (toAddress(selectedOptionTo?.value));
+	}, [currentSolver, isOutputTokenEth, networks, safeChainID, selectedOptionTo?.value]);
+
+	/* 🔵 - Yearn Finance **************************************************************************
+	** SWR hook to get the expected out for a given in/out pair with a specific amount. This hook is
+	** called every 10s or when amount/in or out changes. Calls the allowanceFetcher callback.
+	**********************************************************************************************/
+	const	{data: vanillAllowanceFrom, isLoading: isValidatingAllowance, mutate: mutateAllowance} = useSWR(
+		canInteract && !isInputTokenEth && shouldUseVanillaAllowance ?
+		[selectedOptionFrom, spender] as TAllowanceFetcher : null,
+		retrieveAllowance,
+		{revalidateOnFocus: false}
+	);
+
+	useEffect((): void => {
+		if (shouldUseVanillaAllowance && vanillAllowanceFrom) {
+			set_allowanceFrom(vanillAllowanceFrom);
+		}
+	}, [vanillAllowanceFrom, shouldUseVanillaAllowance]);
+
+	const onSuccess = useCallback(async (): Promise<void> => {
+		onChangeAmount(DefaultTNormalizedBN);
+		await refresh();
+	}, [onChangeAmount, refresh]);
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Trigger an approve web3 action, simply trying to approve `amount` tokens
+	** to be used by the Partner contract or the final vault, in charge of
+	** depositing the tokens.
+	** This approve can not be triggered if the wallet is not active
+	** (not connected) or if the tx is still pending.
+	**************************************************************************/
+	async function	onApproveFrom(): Promise<void> {
+		onApprove(
+			set_txStatusApprove,
+			async (): Promise<void> => {
+				if (currentSolver === Solvers.COWSWAP) {
+					set_allowanceFrom({
+						raw: ethers.constants.MaxUint256,
+						normalized: Infinity
+					});
+				} else {
+					await mutateAllowance();
+				}
+			}
+		);
+	}
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Trigger a deposit web3 action, simply trying to deposit `amount` tokens to
+	** the selected vault.
+	**************************************************************************/
+	async function	onDepositToVault(): Promise<void> {
+		onExecuteDeposit(
+			set_txStatusDeposit,
+			async (): Promise<void> => {
+				await onSuccess();
+			}
+		);
```

Comment:
Can't we just do this?
```suggestion
		onExecuteDeposit(set_txStatusDeposit, onSuccess);
```

Reply:
Nice, thanks for removing those functions. Much cleaner 👍